python/lsst/db/db.py

```diff
@@ -54,547 +41,84 @@
-            return False
-        cmd = "SELECT COUNT(*) FROM information_schema.schemata "
-        cmd += "WHERE schema_name = '%s'" % dbName
-        count = self.execCommand1(cmd)
-        return count[0] == 1
-
-    def dropDb(self, dbName, mustExist=True):
-        """
-        Drop database <dbName>.
-
-        @param dbName     Database name.
-        @param mustExist  Flag indicating what to do if the database does not exist.
-
-        Raise exception if the database does not exist and the flag mustExist is
-        not set to False. Disconnect from the database if it is the current
-        database.
-        """
-        try:
-            self.execCommand0("DROP DATABASE `%s`" % dbName)
-        except DbException as e:
-            if e.errCode() == DbException.DB_DOES_NOT_EXIST and not mustExist:
-                self._log.debug("dropDb failed, mustExist is False")
-            else:
-                raise
-
-    #### Table-related functions ###################################################
-    def tableExists(self, tableName, dbName=None):
-        """
-        Return True if table <tableName> exists in database <dbName>.
-
-        @param tableName  Table name.
-        @param dbName     Database name.
-
-        If <dbName> is not set, the current database name will be used.
-        """
-        dbNameStr = "'%s'" % dbName if dbName is not None else "DATABASE()"
-        cmd = "SELECT COUNT(*) FROM information_schema.tables "
-        cmd += "WHERE table_schema = %s AND table_name = '%s'" % \
-               (dbNameStr, tableName)
-        count = self.execCommand1(cmd)
-        return count[0] == 1
-
-    def createTable(self, tableName, tableSchema, dbName=None, mayExist=False):
-        """
-        Create table <tableName> in database <dbName>.
-
-        @param tableName   Table name.
-        @param tableSchema Table schema starting with opening bracket.
-        @param dbName      Database name.
-        @param mayExist    Flag indicating what to do if the database exists.
-
-        If database <dbName> is not set, and "use <database>" was called earlier,
-        it will use that database.
-        Raise exception if the table already exists and mayExist flag is not say to
-        True.
-        """
-        dbNameStr = "`%s`." % dbName if dbName is not None else ""
-        try:
-            self.execCommand0("CREATE TABLE %s`%s` %s" % \
-                              (dbNameStr, tableName, tableSchema))
-        except  DbException as e:
-            if e.errCode() == DbException.TB_EXISTS and mayExist:
-                self._log.debug("create table failed, mayExist is True")
-            else:
-                raise
+class Db(object):
+    """@brief Thin wrapper around SQLAlchemy. Caches engine."""
 
-    def dropTable(self, tableName, dbName=None, mustExist=True):
+    def __init__(self, logFlag=False, defFile=None, user=None, passwd=None,
+                 host=None, port=None, db=None, query=None):
```

Comment:
Looking at the code it's not quite clear to me what strategy w.r.t. sqlalchemy really is. Do we want to expose it and let people use its full power, or do we want to hide it and only allow features that are useful in our current setup with mysql? We are exposing engine so potentially people are allowed to do almost anything with the engine. But OTOH this factory for creating engines is not very generic to be universally useful. Or is it just an addition to `sqlalchemy.create_engine` which people could still use independently?